ChatCore/Utils/TaskManager.swift

```diff
@@ -11,6 +11,44 @@ import BackgroundTasks
 
 // MARK: Helper class to automatically manage closures by applying various attributes
 final class TaskManager {
+    class TaskCache {
+        typealias Closure = IdentifiableClosure<TaskCompletionResultHandler, Void>
+        
+        private var storage: [Closure: Set<TaskAttribute>] = [:]
+        private let queue = DispatchQueue(label: "com.strv.chatcore.taskcache.queue")
+        
+        public subscript(key: Closure) -> Set<TaskAttribute> {
```

Comment:
why we have internal class to have public methods?

Reply:
Good point! I removed `public` keywords